### PR TITLE
Part 2 - Replaces ContentRegion script with infinite scrolling ScrollRect

### PR DESCRIPTION
**Pull request by sleroux**
Instead of using the ContentRegion script which captures the mouse down/up events manually, I've used a regular ScrollRect component to give us that bouncy scrolling feel. The content area still only ever uses a max of 5 elements while scrolling but always expands the size of the content to fit.

Edit: Added preview below 

list-stuff

### CODE REVIEW

**Comment by sleroux** [comment]
Currently the images go a bit wonky when scrolling because of the async nature of the image loading. I'm going to fix this in a follow up PR. Another nice to have would be to add a 'loading' item view that indicates that there is more in the list. It'd be cool if we could use that spinning icon you made for the single product page as the loading cell. This is a nice to have as I gave it a crack during this PR and found it to be a pain to implement.

**Comment by sleroux** [comment]
Nevermind I've added a fix for the image thing as part of this branch as well as a fix for the wrong product being opened. The image loading still isn't perfect and I'm not a fan of using boolean guard.

File: Assets/Shopify/UIToolkit/Components/RemoteImageLoader.cs [hunk 1]
```diff
@@ -16,12 +16,16 @@ public class RemoteImageLoader : MonoBehaviour {
...
 
+        public bool LoadingInProgress;
+
         private Image _image;
 
         private delegate void RemoteImageLoadedDelegate(Texture2D texture, string error);
 
         private WebImageCache _imageCache = WebImageCache.SharedCache;
 
+        private bool cancelNextLoad;
```
Comment (JakeCataford) [inline Assets/Shopify/UIToolkit/Components/RemoteImageLoader.cs]: not sure what this is for?

File: Assets/Shopify/UIToolkit/Shops/Generic/Classes/MultiProductListItem.cs [hunk 2]
```diff
@@ -27,13 +27,15 @@ public class MultiProductListItem : MonoBehaviour {
 
             var images = (List<Shopify.Unity.Image>)product.images();
             if (images.Count > 0) {
+                if (ImageLoader.LoadingInProgress) {
+                    ImageLoader.CancelPreviousLoad();
```
Comment (JakeCataford) [inline Assets/Shopify/UIToolkit/Shops/Generic/Classes/MultiProductListItem.cs]: oh i see... so if no new images are returned then we need to stop some loading loop?

**Pull request merged**